appservice/src/main/java/nu/yona/server/device/rest/DeviceController.java

```diff
@@ -179,6 +180,15 @@
 		}
 	}
 
+	private static void assertValidDeviceDataForRegister(DeviceRegistrationRequestDto requestDto)
+	{
+		String hint = "All properties are manadatory, except for the FireBase ID";
+		Require.isNonNull(requestDto.name, () -> InvalidDataException.missingProperty("name", hint));
```

Comment:
Can we not put validation JSON deserialization annotation on the `Dto` instead? Anyways, this will work, so I'll approve and merge.

Reply:
I tried marking the property as mandatory, but that didn't work.

Reply:
How about following the approach in [this post](https://www.petrikainulainen.net/programming/spring-framework/spring-from-the-trenches-adding-validation-to-a-rest-api/)?

Reply:
That's indeed an attractive approach. However, currently, we often have dynamic assertions: the old app is allowed more freedom than the new one. Once the users adopted the new app, we can refactor the code. I've created [YD-635](https://jira.yona.nu/browse/YD-635) for this.